### ISSUE TITLE
Style/EmptyClassDefinition casts too wide of a net

### ISSUE THREAD

**Issue by pdobb-n2** [rubocop/rubocop#14961]
## Is your feature request related to a problem? Please describe.

[Style/EmptyClassDefinition](https://docs.rubocop.org/rubocop/latest/cops_style.html#styleemptyclassdefinition) currently requires one to select between:

  ```ruby
  # class_keyword style
  class MyModel < ApplicationRecord  # <--- Should be good
  end
  ```
and:
  ```ruby
  class MyService
    # class_new style
    Error = Class.new(StandardError)  # <--- Should also be good
  end
  ```

Must these be mutually exclusive options? They have different contexts.

## Describe the solution you'd like

Maybe we need a distinction between top-level classes and sublclasses? Could be a separate cop even. But in my above example, I want both to be good. I think what I want to be able to have an empty top-level class, but not an empty subclass.

So, could have Style/EmptyClassDefinition only consider top-level classes. Then could have Style/EmptySubClassDefinition consider only subclasses? (Might prefer renaming Style/EmptyClassDefinition -> Style/EmptyTopLevelClassDefinition as well, for symmetry?)

## Describe alternatives you've considered

I've currently disabled Style/EmptyClassDefinition. Essentially all I really care about here is the ability to define "Error" classes using the `class_new` style, tbh.

## Additional context

I'm currently on Rubocop 1.85.0. Was testing the new changes to Style/EmptyClassDefinition.

---

By the way, I think the [documentation for Style/EmptyClassDefinition](https://docs.rubocop.org/rubocop/latest/cops_style.html#styleemptyclassdefinition) fell out of date in v1.85.0. It still says:

> The supported styles are:
> class_definition (default) - prefer standard class definition over Class.new
> class_new - prefer Class.new over class definition

But I think the top option is meant to be `class_keyword` now.

Also, Configurable Attributes table at the bottom lists 3 options (not 2) under Configurable values. Remove `class_definition` here?

**Comment by hammadxcm** [comment]
I've opened a PR to address this: #14974

It adds an `AllowedParentClasses` configuration option to `Style/EmptyClassDefinition`. When a parent class is listed, both `Class.new(Parent)` and `class Foo < Parent; end` forms are accepted regardless of the enforced style.

Example configuration for your use case:

```yaml
Style/EmptyClassDefinition:
  EnforcedStyle: class_keyword
  AllowedParentClasses:
    - StandardError
```

This also fixes the outdated documentation that still referenced `class_definition` instead of `class_keyword`.

**Comment by pdobb-n2** [comment]
Thank you this works great in v1.86.0!
